GPU/GLES/StateMapping.cpp

```diff
@@ -835,22 +835,19 @@ void TransformDrawEngine::ApplyDrawState(int prim) {
 
 		// This means that to get the analogue glViewport we must:
 		float vpX0 = vpXCenter - offsetX - fabsf(vpXScale);
-		float vpY0 = vpYCenter - offsetY + fabsf(vpYScale);   // Need to account for sign of Y
+		float vpY0 = vpYCenter - offsetY - fabsf(vpYScale);   // Need to account for sign of Y
 		gstate_c.vpWidth = vpXScale * 2.0f;
-		gstate_c.vpHeight = -vpYScale * 2.0f;
+		gstate_c.vpHeight = vpYScale * 2.0f;
 
 		float vpWidth = fabsf(gstate_c.vpWidth);
 		float vpHeight = fabsf(gstate_c.vpHeight);
 
+		// This multiplication should probably be done after viewport clipping. Would let us very slightly simplify the clipping logic?
```

Comment:
Should it?  We clip based on renderWidth/renderHeight, because we stay within the FBO.  It's not based on the framebuffer estimated size.

-[Unknown]